src/gsi/gsi_rfv3io_mod.f90

```diff
@@ -528,14 +562,21 @@ subroutine read_fv3_files(mype)
        in_unit=15
        iwan=0
 !WWWWWW setup for one first guess file for now
-!      do i=0,9 !place holder for FGAT
-       i=3
+       do i=0,99 !place holder for FGAT
+!OU       i=3
+          if (i.eq.6) then
+           write(filenames,"(A11)") 'fv3_dynvars'
+          else
+           write(filenames,"(A12,I2.2)") 'fv3_dynvars_',i
+          endif
+          INQUIRE(FILE=filenames, EXIST=fexist)
+          if(.not.fexist) cycle
 
 !wwww read in from the external file directly, no internal files sigfxx for FV3
           idate5(1)=  regional_time(1)
           idate5(2)=  regional_time(2)
           idate5(3)=  regional_time(3)
-          idate5(4)=  regional_time(4)
+          idate5(4)=  regional_time(4) + i - 6
```

Comment:
This line is supposed to give the hours . However, this item is supposed to be the hour in a day.  The current line may give the number outside of 0-24 .  In that situation, the code can still work?   I am not sure about this.  Hence, some transformation may be needed. I guess there are some function in w3fs21 lib to help on this.

Reply:
Yes, good catch.  Modified using the w3fs15.